### PR TITLE
[14.0][FIX] account_financial_report fix to take the id of the currency if …

### PR DESCRIPTION
**Pull request by matteoopenf**
FIx case the currecny is set inside the account

### CODE REVIEW

**Review by patrickt-oforce — APPROVED** [review]

File: account_financial_report/report/abstract_report.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def _get_accounts_data(self, accounts_ids):
                         "name": account.name,
                         "hide_account": False,
                         "group_id": account.group_id.id,
-                        "currency_id": account.currency_id or False,
+                        "currency_id": account.currency_id.id or False,
```
Comment (lorenzoallegrucci) [inline account_financial_report/report/abstract_report.py]: mi verrebbe da dire che basti `account.currency_id.id` , dato che se il recordset `account.currency_id` è nullo poi `account.currency_id.id` ritorna False se non erro. D'altronde due righe sopra fa `"group_id": account.group_id.id`

Comment (matteoopenf) [reply account_financial_report/report/abstract_report.py]: done

**Review by lorenzoallegrucci — APPROVED** [review]

**Review by amosdisabatino — APPROVED** [review]

**Review by patrickt-oforce — APPROVED** [review]

**Comment by matteoopenf** [comment]
@OCA/accounting-maintainers  is it possibile pmerge?

**Comment by matteoopenf** [comment]
@pedrobaeza  is it possibile merge?
thank you so much for the help

**Comment by pedrobaeza** [comment]
Is this happening due to any error? Can you please include the stack trace in the commit message?

**Comment by matteoopenf** [comment]
> Is this happening due to any error? Can you please include the stack trace in the commit message?

@pedrobaeza  sorry for my delay
if haven't the fix I obtain that error:

```
 opt/odoo/odoo/odoo/models.py", line 3681, in write
    field.write(self, vals[fname])
  File "/opt/odoo/odoo/odoo/fields.py", line 3057, in write
    return self.write_batch([(records, value)])
  File "/opt/odoo/odoo/odoo/fields.py", line 3078, in write_batch
    return self.write_real(records_commands_list, create)
  File "/opt/odoo/odoo/odoo/fields.py", line 3254, in write_real
    flush()
  File "/opt/odoo/odoo/odoo/fields.py", line 3216, in flush
    comodel.create(to_create)
  File "<decorator-gen-69>", line 2, in create
  File "/opt/odoo/odoo/odoo/api.py", line 348, in _model_create_multi
    return create(self, arg)
  File "/opt/odoo/odoo/odoo/addons/base/models/ir_fields.py", line 534, in create
    recs = super().create(vals_list)
  File "<decorator-gen-13>", line 2, in create
  File "/opt/odoo/odoo/odoo/api.py", line 348, in _model_create_multi
    return create(self, arg)
  File "/opt/odoo/odoo/odoo/models.py", line 3886, in create
    records = self._create(data_list)
  File "/opt/odoo/odoo/odoo/models.py", line 3992, in _create
    cr.execute(query, params)
  File "<decorator-gen-3>", line 2, in execute
  File "/opt/odoo/odoo/odoo/sql_db.py", line 101, in check
    return f(self, *args, **kwargs)
  File "/opt/odoo/odoo/odoo/sql_db.py", line 301, in execute
    res = self._obj.execute(query, params)
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo/odoo/odoo/http.py", line 641, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/odoo/http.py", line 317, in _handle_exception
    raise exception.with_traceback(None) from new_cause
psycopg2.ProgrammingError: can't adapt type 'res.currency'
```

with the fix works correctly.
is enough now?
thank you in advance

**Comment by pedrobaeza** [comment]
Traceback seems to be incomplete, and please put it on your commit message for having it auto-documented.

**Comment by matteoopenf** [comment]
Ok, il Will Put tomorrow on commit Matteo ***@***.*** Il giorno 17 apr 2023, alle ore 21:34, Pedro M. Baeza ***@***.***> ha scritto:﻿
Traceback seems to be incomplete, and please put it on your commit message for having it auto-documented.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

Openforce Srls Unipersonale | www.openforce.it | 02 21119472 | Via Tiziano,32 20145 Milano

**Comment by matteoopenf** [comment]
> Traceback seems to be incomplete, and please put it on your commit message for having it auto-documented.

done

**Comment by pedrobaeza** [comment]
Thanks

/ocabot merge patch

Please fw-port it if applicable

**Comment by matteoopenf** [comment]
> Thanks
> 
> /ocabot merge patch
> 
> Please fw-port it if applicable

https://github.com/OCA/account-financial-reporting/pull/1030 I made the forwar the fix 
thanks again @pedrobaeza

**Pull request merged**